examples/monitor/main.go

```
@@ -0,0 +1,125 @@
+		Size:    24,
+		DPI:     dpi,
+		Hinting: font.HintingFull,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+const (
+	screenWidth  = 640
+	screenHeight = 480
+)
+
+type Game struct {
+}
+
+func (g *Game) Update() error {
+	if inpututil.IsMouseButtonJustPressed(ebiten.MouseButton0) {
+		cx, cy := ebiten.CursorPosition()
+		l := text.BoundString(mplusNormalFont, "|").Dy()
+		y := l
+
+		b := text.BoundString(mplusNormalFont, "toggle fullscreen")
+		if cx >= b.Min.X && cx <= b.Max.X && cy >= b.Min.Y+y && cy <= b.Max.Y+y {
+			ebiten.SetFullscreen(!ebiten.IsFullscreen())
+			return nil
+		}
+		y += l
+
+		for i, m := range ebiten.AppendMonitors(nil) {
+			b := text.BoundString(mplusNormalFont, fmt.Sprintf("%d: %s %s", i, m.Name(), m.Bounds().String()))
+			if cx >= b.Min.X && cx <= b.Max.X && cy >= b.Min.Y+y && cy <= b.Max.Y+y {
+				ebiten.SetWindowTitle(m.Name())
+				ebiten.SetWindowMonitor(m)
+				break
+			}
+			y += l
+		}
+	}
+	return nil
+}
+
+func (g *Game) Draw(screen *ebiten.Image) {
+	const x = 0
+	l := text.BoundString(mplusNormalFont, "|").Dy()
+	y := l
+	text.Draw(screen, "toggle fullscreen", mplusNormalFont, x, y, color.White)
+	y += l
+	for i, m := range ebiten.AppendMonitors(nil) {
+		text.Draw(screen, fmt.Sprintf("%d: %s %s", i, m.Name(), m.Bounds().String()), mplusNormalFont, x, y, color.White)
+		y += l
+	}
+
+	activeMonitor := ebiten.WindowMonitor()
+	for i, m := range ebiten.AppendMonitors(nil) {
+		if m == activeMonitor {
+			text.Draw(screen, fmt.Sprintf("active: %s (%d)", m.Name(), i), mplusNormalFont, x, y, color.White)
+		}
+	}
+}
+
+func (g *Game) Layout(outsideWidth, outsideHeight int) (int, int) {
+	return screenWidth, screenHeight
+}
+
+func main() {
+	g := &Game{}
+
+	// Spawn the window on the last monitor reported.
+	var targetMonitor *ebiten.Monitor
+	monitors := ebiten.AppendMonitors(nil)
+	if len(monitors) > 0 {
+		targetMonitor = monitors[len(monitors)-1]
+	}
+
+	ebiten.SetWindowMonitor(targetMonitor)
```

Is this expected to pass a nil value to `SetWindowMonitor`?

---

Added a check for SetWindowMonitor to return if nil is passed. This would have caused issues.

---

Then, this program also should abort before passing nil here.

---

I changed it to just silently return in `ebiten.SetWindowMonitor` if it is nil. Would it be better to panic?

---

panic is better.

---

Adjusted to be:
```go
func (w *glfwWindow) SetMonitor(monitor *Monitor) {
	if monitor == nil {
		panic("ui: SetMonitor can't be nil")
	}
```

---

> ui: SetMonitor can't be nil

ui: monitor cannot be nil at SetMonitor